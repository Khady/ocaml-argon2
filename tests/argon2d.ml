open Argon2
open Printf

let hash_len = 32

let t_cost = 2

let m_cost = 65536

let parallelism = 1

let salt = "0000000000000000"

let salt_len = String.length salt

let pwd = "password"

let hash_raw () =
  match D.hash_raw ~t_cost ~m_cost ~parallelism ~pwd ~salt ~hash_len with
  | Result.Ok hash ->
      printf "argon2d hash:";
      D.hash_to_string hash
      |> String.iter (fun c -> Printf.printf "%02x" (Char.code c));
      printf "\n"
  | Result.Error e ->
      printf "Error while computing hash with argon2d_hash_raw: %s\n"
        (ErrorCodes.message e)

let () =
  hash_raw ();
  ()
