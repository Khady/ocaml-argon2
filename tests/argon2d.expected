argon2d hash:30f5e9f2584e962bba213cc256dcf9ccdce18d8f67a6cbf8e012e0619b6d52d2
argon2d encoded:$argon2d$v=19$m=65536,t=2,p=1$MDAwMDAwMDAwMDAwMDAwMA$MPXp8lhOliu6ITzCVtz5zNzhjY9npsv44BLgYZttUtI
